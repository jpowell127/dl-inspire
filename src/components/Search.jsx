import React from "react";
import PropTypes from "prop-types";
import { Input, Button } from "semantic-ui-react";

export default function Search({
  getUsers,
  handleSearchTermChange,
  onKeyPress
}) {
  return (
    <Input
      inverted
      size="large"
      action={{
        color: "pink",
        icon: "search",
        onClick: getUsers
      }}
      placeholder="Search Behance"
      onChange={handleSearchTermChange}
      onKeyPress={onKeyPress}
    />
  );
}

Search.propTypes = {
  getUsers: PropTypes.func.isRequired,
  handleSearchTermChange: PropTypes.func.isRequired
};
