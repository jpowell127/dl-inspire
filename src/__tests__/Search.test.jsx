import React from "react";
import Search from "../components/Search";
import SearchProps from "./mocks/mock.searchProps";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

const props = SearchProps;

describe("<Search />", () => {
  it("renders correctly", () => {
    const wrapper = mount(<Search {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("it calls getUsers when search button is clicked", () => {
    const wrapper = mount(<Search {...props} />);
    wrapper
      .find("input")
      .simulate("change", { currentTarget: { value: "test" } });
    wrapper.find("button").simulate("click");
    expect(props.getUsers).toHaveBeenCalled();
  });
  it("it calls getUsers when enter button is pressed", () => {
    const wrapper = mount(<Search {...props} />);
    wrapper
      .find("input")
      .simulate("change", { currentTarget: { value: "test" } });
    wrapper.find("button").simulate("keypress", { key: "Enter" });
    expect(props.getUsers).toHaveBeenCalled();
  });
});
