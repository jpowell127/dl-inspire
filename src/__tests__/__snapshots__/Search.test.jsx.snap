// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<Search
  getUsers={[MockFunction]}
  handleSearchTermChange={[MockFunction]}
  onKeyPress={[MockFunction]}
>
  <Input
    action={
      Object {
        "color": "pink",
        "icon": "search",
        "onClick": [MockFunction],
      }
    }
    inverted={true}
    onChange={[MockFunction]}
    onKeyPress={[MockFunction]}
    placeholder="Search Behance"
    size="large"
    type="text"
  >
    <div
      className="ui large inverted action input"
    >
      <input
        key="text"
        onChange={[Function]}
        onKeyPress={[MockFunction]}
        placeholder="Search Behance"
        type="text"
      />
      <Button
        as="button"
        color="pink"
        icon="search"
        onClick={[MockFunction]}
        role="button"
      >
        <button
          className="ui pink icon button"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            key="search"
            name="search"
          >
            <i
              aria-hidden="true"
              className="search icon"
            />
          </Icon>
        </button>
      </Button>
    </div>
  </Input>
</Search>
`;
